plugins/notifications-frontend/src/components/NotificationsSidebarItem.tsx

```diff
@@ -0,0 +1,140 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React from 'react';
+
+import { SidebarItem } from '@backstage/core-components';
+import { useApi } from '@backstage/core-plugin-api';
+
+import {
+  IconButton,
+  Link,
+  makeStyles,
+  Snackbar,
+  Tooltip,
+} from '@material-ui/core';
+import CloseIcon from '@material-ui/icons/Close';
+import NotificationsIcon from '@material-ui/icons/Notifications';
+import NotificationsOffIcon from '@material-ui/icons/NotificationsOff';
+
+import { notificationsApiRef } from '../api';
+import { NOTIFICATIONS_ROUTE } from '../constants';
+import { Notification } from '../openapi';
+import { usePollingEffect } from './usePollingEffect';
+
+const NotificationsErrorIcon = () => (
+  <Tooltip title="Failed to load notifications">
+    <NotificationsOffIcon />
+  </Tooltip>
+);
+
+export type NotificationsSidebarItemProps = {
+  /**
+   * Number of milliseconds between polling the notifications backend.
+   * If negative or zero, the poling is not started.
+   * Example: 5000
+   */
+  pollingInterval?: number;
+};
+
+const useStyles = makeStyles(_theme => ({
+  systemAlertAction: {
+    marginRight: '1rem',
+  },
+}));
+
+export const NotificationsSidebarItem = ({
+  pollingInterval,
+}: NotificationsSidebarItemProps) => {
+  const styles = useStyles();
+  const notificationsApi = useApi(notificationsApiRef);
+
+  const [error, setError] = React.useState<Error | undefined>(undefined);
+  const [unreadCount, setUnreadCount] = React.useState(0);
+  const [pageLoadingTime] = React.useState(new Date(Date.now()));
+  const [lastSystemWideNotification, setLastSystemWideNotification] =
+    React.useState<Notification>();
+  const [closedNotificationId, setClosedNotificationId] =
+    React.useState<string>();
+
+  const pollCallback = React.useCallback(async () => {
+    try {
+      setUnreadCount(
+        await notificationsApi.getNotificationsCount({
+          read: false,
+          messageScope: 'user',
+        }),
+      );
+
+      const data = await notificationsApi.getNotifications({
+        pageSize: 1,
+        pageNumber: 1,
+        createdAfter: pageLoadingTime,
+        orderBy: 'created',
+        orderByDirec: 'desc',
+        messageScope: 'system',
+      });
+
+      setLastSystemWideNotification(data?.[0]);
+    } catch (e: unknown) {
+      setError(e as Error);
+    }
+  }, [notificationsApi, pageLoadingTime]);
+
+  usePollingEffect(pollCallback, [], pollingInterval);
+
+  let icon = NotificationsIcon;
+  if (!!error) {
+    icon = NotificationsErrorIcon;
+  }
+
+  return (
+    <>
+      <SidebarItem
+        icon={icon}
+        to={NOTIFICATIONS_ROUTE}
```

Comment:
You should use `useRouteRef(rootRouteRef)` instead hard-coding the NOTIFICATIONS_ROUTE

Reply:
done, thanks